### DISCUSSION TITLE
Citi Hackathon FDC3 Use Case: Create FDC3 developer tools

### DISCUSSION THREAD

**Discussion by lucaborella89** [finos/finos-pmcs#330] [Ideas]
# Create FDC3 developer tools
<img width="794" alt="fdc3-logo-2019-color" src="https://github.com/user-attachments/assets/a660af2f-41f9-4cdc-819a-62e9dc4e5dd3">

## Expected Features
FDC3 Workbench has been a great help for developers getting started with FDC3 and developing their first app integrations https://fdc3.finos.org/toolbox/fdc3-workbench/ (needs to be run inside a desktop agent). 
However, it takes up a lot of the screen - hence there is scope for FDC3 developer tools with other form factors, e.g.:
1. Widgets that can: Display context received on a channel
2. Broadcast context on a channel on a button push. Raise an intent on a button push
3. Integrate with test frameworks to send or receive context or intents as part of a test etc.

## Useful info 
Kris West is the maintainer of FDC3
Alexandru Dumitru is a contributor and Project Administrator to FDC3
See: https://github.com/finos/FDC3/blob/main/toolbox/fdc3-workbench/README.md

**Comment by kriswest** [comment]
Please note that to work with the FDC3 API you need access to an [FDC3 Desktop Agent](https://fdc3.finos.org/docs/api/spec), which will make the API available to your app. There is a web sandbox, made available by [interop.io](https://interop.io/) that provides an FDC3 Desktop Agent for hackathon participants at: https://citi-fdc3.cloud.interop.io/ (update expected by EOD today). 

There will be an introduction session presented by @julianna-ciq from interop.io on the sandbox and using the FDC3 API in your apps on the 28th October. You can also checkout recent introductions to the FDC3 API at [FINOS' OSFF conference](https://events.linuxfoundation.org/open-source-finance-forum-new-york/) from: 
- @robmoffat: https://www.youtube.com/watch?v=9uUKTk2e-Xs
- @julianna-ciq: https://www.youtube.com/watch?v=D6WST19pYUs

Don't be afraid to go beyond the brief on this use case - be creative and see what you can do with the FDC3 API, example applications are themselves great development tools in interop as app developers always need other apps to talk to!
